anvil-python/anvil/commands/haproxy.py

```diff
@@ -34,11 +38,48 @@
 HAPROXY_UNIT_TIMEOUT = (
     1200  # 15 minutes, adding / removing units can take a long time
 )
+HAPROXY_CERTS_DIR = os.path.join(environ['HOME'], "certs/")
+LOG = logging.getLogger(__name__)
+
+
+def validate_cert_file(filepath: str) -> None:
+    if not os.path.isfile(filepath):
+        raise ValueError(f"{filepath} does not exist")
+    with open(filepath, "r") as f:
+        if "BEGIN CERTIFICATE" not in f.read():
+            raise ValueError("Invalid certificate file")
+
+def validate_key_file(filepath: str) -> None:
+    if not os.path.isfile(filepath):
+        raise ValueError(f"{filepath} does not exist")
+    with open(filepath, "r") as f:
+        if "BEGIN PRIVATE KEY" not in f.read():
+            raise ValueError("Invalid key file")
 
 
 class DeployHAProxyApplicationStep(DeployMachineApplicationStep):
     """Deploy HAProxy application using Terraform"""
 
+    _HAPROXY_QUESTIONS = {
+        "ssl_cert": questions.PromptQuestion(
+            "Path to SSL Certificate for HAProxy: ",
+            validation_function=validate_cert_file,
+        ),
+        "ssl_key": questions.PromptQuestion(
+            "Path to private key for the SSL certificate",
+            validation_function=validate_key_file,
+        ),
+    }
+    _TLS_SERVICES_CONFIG = f"""- service_name: haproxy_service
+  service_host: "0.0.0.0"
+  service_port: 443
+  service_options:
+    - balance leastconn
```

Comment:
What is the non-related TLS configuration if we do not setup TLS? How can we guarantee that those options do not afffect the HAProxy setup?

Reply:
The only changes to the `services.yaml` for TLS termination are the `service_port` and `crts` keys. If the user opts to skip TLS configuration, the charm's default `services.yaml` is used